nslsii/__init__.py

```diff
@@ -471,3 +475,40 @@ def migrate_metadata():
     os.makedirs(directory, exist_ok=True)
     new_md = PersistentDict(directory)
     new_md.update(old_md)
+
+
+def subscribe_kafka_publisher(RE, beamline_name, bootstrap_servers):
+    """
+    Create and subscribe a Kafka Publisher to a RunEngine. Keep a reference to the Publisher.
+    The Publisher will publish to Kafka topic "<beamline>.bluesky.documents".
+
+    Parameters
+    ----------
+    RE: RunEngine
+        the RunEngine to which the Kafka Publisher will be subscribed
+
+    beamline_name: str
+        beamline name, for example "csx", to be used in building the
+        Kafka topic to which messages will be published
+
+    bootstrap_servers: str
+        Comma-delimited list of Kafka server addresses as a string such as ``'10.0.137.8:9092'``
+
+    Returns
+    -------
+    subscription_token: int
+        use this to unsubscribe the Publisher from the RE this way: ``RE.unsubscribe(subscription_token)``
+
+    """
+    topic = f"{beamline_name.lower()}.bluesky.documents"
+    _kafka_publisher = Publisher(
+         topic=topic,
+         bootstrap_servers=bootstrap_servers,
+         key=uuid.uuid4(),
+         producer_config={
+             "enable.idempotence": True
+         }
+    )
+    subscription_token = RE.subscribe(_kafka_publisher)
+    print(f'RE will publish documents to Kafka topic {topic}')
```

Comment:
Questionable whether we want this to print at `IPython` startup.

Reply:
I will log it instead.